### PR TITLE
fix: remark excerpt add halfChop option

### PR DESCRIPTION
**Pull request by youngboy**
Default prune won't work for non-english text.  Add `halfChop` (default to true) option, set it to false so that non-english blogs will make it fallback to lodash `truncate`. 

I think this resolve https://github.com/gatsbyjs/gatsby/issues/6916

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit f6325a1cdfb18a32800baac981d5123d495e95be

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6476f573f2cf60d184734e

**Comment by pieh** [comment]
Could we detect this automatically? that prune is failing and automatically switch to `truncate`?

If not then this looks good, not sure about `halfChop` name - is this reference to something that I don't understand? :) Please document this option in readme, and possibly add to troubleshooting section for "excerpts for non-latin languages" or something like that?

**Comment by youngboy** [comment]
@pieh So sorry about confusing name `halfChop`, I 'borrow' this from [underscore.string's readme](https://github.com/epeli/underscore.string#prunestring-length-prunestring--string)

> prune(string, length, pruneString) => string
>
> Elegant version of truncate. Makes sure the pruned string does not exceed the original length. Avoid half-chopped words when truncating.

How about just rename it to `truncate` ?

For detect failing case automatically, I think there is no straightforward way.

**Comment by pieh** [comment]
> How about just rename it to truncate ?
Let's go with that. Most important parts is to add docs for this

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit c02962268dc52c7235c5e27f951d00e1f751c20a

https://app.netlify.com/sites/gatsbyjs/deploys/5b687301e39e7c5e43b60c7e

File: packages/gatsby-transformer-remark/README.md [hunk 1]
```diff
@@ -63,3 +63,20 @@ A sample GraphQL query to get MarkdownRemark nodes:
   }
 }
 ```
+
+## Troubleshooting
+
+### Excerpts for non-latin languages
+
+By default, excerpts is avoid half-chopped words when truncating content.
```
Comment (pieh) [inline packages/gatsby-transformer-remark/README.md]: How about something like this:

> By default, `excerpt` uses `underscore.string/prune` which doesn't handle non-latin characters (https://github.com/epeli/underscore.string/issues/418). 
>
> If that is the case, you can set `truncate` option on `excerpt` field, like:

and your query after that?

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit 12e87bde3392b730d68f384f1b749282e87fb0e9

https://app.netlify.com/sites/using-contentful/deploys/5b762897b312747207864f98

File: packages/gatsby-transformer-remark/README.md (hunk 1 above)
Comment (youngboy) [reply packages/gatsby-transformer-remark/README.md]: Thanks 👍, updated.

**Pull request merged**